doc/tutorial/conv_arithmetic.txt

```diff
@@ -0,0 +1,597 @@
+of kernel size, stride and zero padding along a given axis only affect the
+output size of that axis, which means we can treat axes separately in computing
+the output shape.
+
+.. note::
+
+    For the remainder of this tutorial, we will concentrate on 2D convolutions
+    mapping one input feature map to one output feature map. The relationships
+    presented here also hold in the general case, i.e. an N-dimensional
+    convolution with arbitrary input feature maps, input shape, kernel shape,
+    output feature maps, zero padding and strides.
+
+    Since height and width are treated intependently in computing a
+    convolution's output shape, we will also restrict ourselves to square input,
+    kernel, stride and padding shapes.
+
+Simplest case
+=============
+
+Let's start with the simplest case, where a kernel just slides across the input:
+
+* input size :math:`i`,
+* kernel size :math:`k`,
+* stride :math:`s = 1`,
+* padding size :math:`p = 0` (``border_mode='valid'`` in Theano).
+
+Here is what the convolution looks like for :math:`i = 4` and :math:`k = 3`:
+
+.. figure:: conv_arithmetic_figures/no_padding_no_strides.gif
+
+One way of defining the output size is by the number of possible placements of
+the kernel on the input. Let's consider the x-axis: the left side of the kernel
+starts at the left side of the input, and we slide the kernel until its right
+side touches the right side of the input. The size of the output will be equal
+to the number of steps we need to make, plus one, because we need to account for
+the initial position of the kernel. The same applies for the y-axis.
+
+From this, we conclude that the output size :math:`o` of the convolution is
+
+.. math::
+
+    o = (i - k) + 1.
+
+Zero padding
+============
+
+If we factor in zero padding, i.e.
+
+* input size :math:`i`,
+* kernel size :math:`k`,
+* stride :math:`s = 1`,
+* padding size :math:`p`,
+
+all we need to do is to apply the previous rule on the augmented input. Since
+padding with :math:`p` zeros adds :math:`2p` to the input size (:math:`p` on
+each side), the output size :math:`o` of the convolution becomes
+
+.. math::
+
+    o = (i - k) + 2p + 1.
+
+Here is an example for :math:`i = 5`, :math:`k = 4` and :math:`p = 2`:
+
+.. figure:: conv_arithmetic_figures/arbitrary_padding_no_strides.gif
+
+Special cases
+-------------
+
+There are two special cases of zero padding to take note of, because they're
+quite useful in practice.
+
+Half
+^^^^
+
+If :math:`k` is odd (:math:`k = 2n + 1`) and
+:math:`p = \lfloor k / 2 \rfloor = n`, we have that
+
+.. math::
+
+    \begin{split}
+    o &= i + 2 \lfloor k / 2 \rfloor - (k - 1) \\
+      &= i + 2n - 2n \\
+      &= i.
+    \end{split}
+
+In other words, **the output size is equal to the input size**. This is
+sometimes referred to as ``half`` (or ``same``) padding. Here is an example for
+:math:`i = 5`, :math:`k = 3` and (therefore) :math:`p = 1`:
```

Comment:
Break the line before "Here", as the final colon goes in the new line by its own otherwise.